core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PreferWritePartitioning.java

```diff
@@ -46,6 +49,28 @@ public Result apply(TableWriterNode node, Captures captures, Context context)
             return Result.empty();
         }
 
+        int minimumNumberOfPartitions = getPreferredWritePartitioningMinNumberOfPartitions(context.getSession());
+        if (minimumNumberOfPartitions == 0) {
```

Comment:
Make it `<= 1`. I would say if somebody sets min to `1`, then preferred partitioning should always be used since one could pretty much assume insert will write to at least one partition